packages/lib/src/components/PaywallRenderer.svelte

```diff
@@ -30,11 +31,36 @@
 
   let { t, horizontal = false, paywall }: Props = $props();
 
-  const { subscriptions, currency, features } = paywall;
+  const {
+    subscriptions,
+    currency,
+    features,
+    settings: {
+      styling: { showBackground, dropShadow, backgroundColor }
+    }
+  } = paywall;
+
+  const paywallBgColor = hexToHsl(backgroundColor || '#F6DFDC');
+
+  let sesamyPaywallDesignTokens = `
+    .base {
+      --s-paywall-bg-start-color: var(--sesamy-paywall-bg-start-color, ${paywallBgColor[0]},${paywallBgColor[1]}%,${paywallBgColor[2]}%);
+      --s-paywall-bg-end-color: var(--sesamy-paywall-bg-end-color, ${paywallBgColor[0]},${paywallBgColor[1]}%,${paywallBgColor[2] + (100 - paywallBgColor[2]) * 0.5}%);
```

Comment:
In detail this sets the `h`, `s`, `l` in hsl.

"`${paywallBgColor[2] + (100 - paywallBgColor[2]) * 0.5}%`" just takes the difference between it's current value and all the way to 100% (white). Then the `0.5` indicates that we want it to be 50% of whatever the difference is.

Again, this is a mouthful but probably the best way to go about it.

Client's will have to specify:
```
---sesamy-paywall-bg-start-color: hsl(x, y%, z%);
---sesamy-paywall-bg-end-color: hsl(x, y%, z%);
```
if they want to override the colors directly.

In portal just setting a "background hex-color of paywall" will do this automatically for them.